src/client/extension.ts

```diff
@@ -192,13 +206,40 @@ export async function activate(
     }
 
     _telemetry.sendTelemetryEvent("activated");
+}
 
+export function setupAuthFileWatcher() {
+    const watchPath = GetAuthProfileWatchPattern();
+    if (watchPath) {
+        const watcher = vscode.workspace.createFileSystemWatcher(watchPath);
+        watcher.onDidChange(() => handleOrgChange());
+        watcher.onDidCreate(() => handleOrgChange());
+        watcher.onDidDelete(() => handleOrgChange());
+    }
+}
+
+async function handleOrgChange() {
+    const cliContext = new CliAcquisitionContext(_context, _telemetry);
+    const cli = new CliAcquisition(cliContext);
+    const cliPath = await cli.ensureInstalled();
+    const pacContext = new PacWrapperContext(_context, _telemetry);
+    const interop = new PacInterop(pacContext, cliPath);
+    const pacWrapper = new PacWrapper(pacContext, interop);
+    const pacActiveOrg = await pacWrapper?.activeOrg();
+    if (pacActiveOrg && pacActiveOrg.Status === PAC_SUCCESS) {
+        const orgID = pacActiveOrg.Results.OrgId;
+        const artemisResponse = await fetchArtemisResponse(orgID, _telemetry);
+        if (artemisResponse) {
+            const { geoName } = artemisResponse[0];
+            oneDSLoggerWrapper.instantiate(geoName);
+            oneDSLoggerWrapper.getLogger().traceInfo(telemetryEventNames.DESKTOP_EXTENSION_INIT_CONTEXT, pacActiveOrg.Results);
+        }
+    }
```

Comment:
Duplicated code from `src/common/copilot/PowerPagesCopilot.ts.`
- Filewatcher can be limited by the OS, so we should generally not have duplicates for the same file
- PAC calls are generally slow (and occur serially in a queue) so we should not be multiple calls for the same information

Reply:
This comment has been addressed here -[https://github.com/microsoft/powerplatform-vscode/pull/826#discussion_r1484632782]( https://github.com/microsoft/powerplatform-vscode/pull/826#discussion_r1484632782)

The changes to address this comment in pr- https://github.com/microsoft/powerplatform-vscode/pull/826 will be merged first - then will take a pull to current branch - post that will check this in.

Reply:
This comment has been addressed. Changes were merged with pr- https://github.com/microsoft/powerplatform-vscode/pull/826.
Latest main has been merged in this branch.